src/applyMiddleware.js

```diff
@@ -24,7 +24,7 @@ export default function applyMiddleware(...middlewares) {
 
     const middlewareAPI = {
       getState: store.getState,
-      dispatch: (action) => dispatch(action)
+      dispatch: (...args) => dispatch(...args)
```

Comment:
why not just return `dispatch`?

```js
const middlewareAPI = {
  getState: store.getState,
  dispatch,
}
```

Reply:
Hey! Can someone have an explanation for this change?

Reply:
@FranciscoMSM This particular change is to allow middleware to change the arguments of dispatch and pass that through. 

The reason for the extra function is to remove direct access to `dispatch` so middleware doesn't try to attach properties or other hidden metadata to it. Each middleware gets its own unique copy of `dispatch` this way, and there is less chance of interference between middlewares.